source/jormungandr/jormungandr/utils.py

```diff
@@ -907,3 +910,37 @@ def get_weekday(timestamp):
 
 def is_stop_point(uri):
     return uri.startswith("stop_point") if uri else False
+
+
+def read_best_boarding_positions(file_path):
+    if not os.path.exists(file_path):
+        return None
+    try:
+        my_dict = dict()
+        fieldnames = ['from_id', 'to_id', 'positionnement_navitia']
+        with open(file_path) as f:
+            csv_reader = csv.DictReader(f, fieldnames)
+            # skip the header
+            next(csv_reader)
+
+            for line in csv_reader:
+                key = BEST_BOARDING_POSITION_KEY.format(line['from_id'], line['to_id'])
+                value = line['positionnement_navitia']
+                my_dict.setdefault(key, []).append(value)
+
+        return my_dict
+    except:
+        logger = logging.getLogger(__name__)
+        logger.exception('Error while loading best_boarding_positions file: {}'.format(file_path))
+        return None
+
+
+def get_best_boarding_position(section, positions_map):
+    if not positions_map:
+        return None
+    if section.type == response_pb2.TRANSFER:
+        key = BEST_BOARDING_POSITION_KEY.format(section.origin.uri, section.destination.uri)
+        return positions_map.get(key, None)
+    elif section.type == response_pb2.STREET_NETWORK and hasattr(section, 'vias') and section.vias:
+        key = BEST_BOARDING_POSITION_KEY.format(section.origin.uri, section.vias[0].uri)
```

Comment:
be careful about `vias[0]`, the public transit access point may not be at position 0 in `vias`. For example I think there may be a [poi_accesss_point](https://github.com/hove-io/navitia/pull/3978) at `vias[0]` and a pt_access_point at `vias[1]` ?
Poke @azime

Reply:
I modified the code to iterate all vias and calculate best boarding position.